main/src/main/scala/sbt/Defaults.scala

```diff
@@ -518,9 +518,9 @@ object Defaults extends BuildCommon {
     ))
 
   lazy val projectTasks: Seq[Setting[_]] = Seq(
-    cleanFiles := cleanFilesTask.value,
+    cleanFiles := Seq(managedDirectory.value, target.value),
     cleanKeepFiles := historyPath.value.toVector,
-    clean := (Def.task { IO.delete(cleanFiles.value) } tag (Tags.Clean)).value,
+    clean := (Def.task { IO.delete(cleanFilesTask.value) } tag (Tags.Clean)).value,
```

Comment:
I think the existing implementation works as designed.
I think it's more useful to have definitive task that returns the list of files to delete, that we can use as dry run.

Reply:
I'm not sure I get you right. Do you say that this solution breaks some intentional design or that there is no issue at all? (i.e. if you override `cleanFiles` and `cleanKeepFiles` and those get deleted anyway, it's normal)

Reply:
Problem with this implementation is that I lose the ability to preview what's about to be cleaned.
Whoever overriding `cleanFiles` needs to be responsible for keeping the existing behavior.